src/twisted/test/test_ssl.py

```diff
@@ -667,8 +671,10 @@ def test_method(self):
         # SSLv23_METHOD allows SSLv2, SSLv3, or TLSv1
         self.assertEqual(self.context._method, SSL.SSLv23_METHOD)
 
-        # And OP_NO_SSLv2 disables the SSLv2 support.
-        self.assertTrue(self.context._options & SSL.OP_NO_SSLv2)
+        # OpenSSL >= 1.1.0 no longer provides SSLv2
+        if not OPENSSL_1_1:
```

Comment:
Sorry, that was a bit terse.  You can compile prior versions of OpenSSL without SSLv2 support, but `SSL_OP_NO_SSLv2` will continue to be defined as some non-zero value.  [In 1.0.2ish, for example, there are no `ifdef`s around its definition](https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/ssl/ssl.h#L644). 

That means this test could pass even on versions that lacked SSLv2 support, because it was possible to set the `SSL_OP_NO_SSLv2` flag and recover it from the context's options.

In 1.1.0+, however, `SSL_OP_NO_SSLv2` is 0.  That means it can be set but it cannot even be checked because it has no effect!

That's why I put the OpenSSL version check here.  We _definitely_ want to ensure we're excluding SSLv2 on earlier versions of OpenSSL where it's even at all possible.